internal/common/tree.go

```diff
@@ -0,0 +1,201 @@
+package common
+
+import (
+	"slices"
+
+	"sigs.k8s.io/external-dns/endpoint"
+
+	"github.com/kuadrant/dns-operator/api/v1alpha1"
+)
+
+type DNSTreeNode struct {
+	Name     string
+	Children []*DNSTreeNode
```

Comment:
maybe we could comment on these to say what they are as it is not hugely clear with the current names

Reply:
like what is a dataset in this struct

Reply:
added comments on structs